libs/api/domains/auth/src/lib/meDelegations.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@nestjs/common'
+
+import { Auth, AuthMiddleware, User } from '@island.is/auth-nest-tools'
+import {
+  DelegationDTO,
+  UpdateDelegationScopeDTO,
+  MeDelegationsControllerFindAllDirectionEnum,
+  MeDelegationsApi,
+} from '@island.is/clients/auth-public-api'
+
+import {
+  UpdateDelegationInput,
+  DeleteDelegationInput,
+  DelegationInput,
+  CreateDelegationInput,
+} from './dto'
+import { DelegationByOtherUserInput } from './dto/delegationByOtherUser.input'
+
+@Injectable()
+export class MeDelegationsService {
+  constructor(private delegationsApi: MeDelegationsApi) {}
+
+  private delegationsApiWithAuth(auth: Auth) {
+    return this.delegationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  getDelegations(user: User): Promise<DelegationDTO[]> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindAll({
+      direction: MeDelegationsControllerFindAllDirectionEnum.Outgoing,
+      isValid: true,
+    })
+  }
+
+  getDelegationById(
+    user: User,
+    { delegationId }: DelegationInput,
+  ): Promise<DelegationDTO> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerFindOne({
+      delegationId,
+    })
+  }
+
+  async getDelegationByOtherUser(
+    user: User,
+    { toNationalId }: DelegationByOtherUserInput,
+  ): Promise<DelegationDTO> {
+    const delegations = await this.delegationsApiWithAuth(
+      user,
+    ).meDelegationsControllerFindAll({
+      direction: MeDelegationsControllerFindAllDirectionEnum.Outgoing,
+      otherUser: toNationalId,
+    })
+
+    return delegations[0]
+  }
+
+  createDelegation(
+    user: User,
+    { toNationalId, scopes }: CreateDelegationInput,
+  ): Promise<DelegationDTO> {
+    return this.delegationsApiWithAuth(user).meDelegationsControllerCreate({
+      createDelegationDTO: { toNationalId, scopes },
```

Comment:
Do the POST/PUT APIs accept data nested in a createDelegationDTO property? Does that follow the Rest Design Guide?

Reply:
Strictly the scopes should be a sub-resource of the delegations. But we might want to rethink the delegation model a bit to simplify this resource definitions.

Leaving as is for now.